db.php

```diff
@@ -1149,6 +1149,16 @@ public function has_cap( $db_cap, $dbh_or_table = false ) {
 	 * @return false|string false on failure, version number on success
 	 */
 	public function db_version( $dbh_or_table = false ) {
+		$server_info = $this->db_server_info( $dbh_or_table );
+		return $server_info ? preg_replace( '/[^0-9.].*/', '', $server_info ) : false;
```

Comment:
WP Core returns `null` when `db_version()` fails: https://github.com/WordPress/WordPress/blob/9c3b7e5e7c94f3f8c5f3766a0d25e801b39a81a5/wp-includes/class-wpdb.php#L3852. Shouldn't we do the same? Backwards-compatibility kept me from making the change, but forward-compatibility with WP seems like just as important a goal?

Reply:
Returning `null` makes sense for forward-compatibility, especially if there are future checks for that value.